content-audit/api/src/main/java/org/commonjava/propulsor/content/audit/model/FileEvent.java

```diff
@@ -18,11 +18,12 @@
 import com.fasterxml.jackson.annotation.JsonFormat;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
+import java.io.Serializable;
 import java.util.Date;
 import java.util.Map;
 import java.util.UUID;
 
-public class FileEvent
+public class FileEvent implements Comparable<FileEvent>, Serializable
```

Comment:
We need to implement Externalizable here, and use a version field to retain control over versioning and serdes (serialization / deserialization). Take a look at `TrackedContentEntry` in Indy for an example of how this almost caused a big problem for us.

**Also, looking at this more, I think we should consider moving this into auditquery, then importing this api from there into Indy.**